libmiemu/controller.h

```diff
@@ -1,8 +1,35 @@
 #ifndef CONTROLLER_H
 #define CONTROLLER_H
 
+#include "datapath/signals.h"
+
+#define ADDI 0000
```

Comment:
Also, you need to prefix all of these with `0b` so they aren't interpreted as octal constants (explains the warning above).